File: docs/snippets/all/archetypes/transform3d_simple.cpp
// Log different transforms between three arrows.

#include <dalaran.hpp>

constexpr float TAU = 6.28318530717958647692528676655900577f;

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_transform3d");
    rec.spawn().exit_on_failure();

    auto arrow = dalaran::Arrows3D::from_vectors({{0.0f, 1.0f, 0.0f}})
                     .with_origins({{0.0f, 0.0f, 0.0f}});

    rec.log("base", arrow);

    rec.log(
        "base/translated",
        dalaran::Transform3D::from_translation({1.0f, 0.0f, 0.0f})
    );
    rec.log("base/translated", arrow);

    rec.log(
        "base/rotated_scaled",
        dalaran::Transform3D::from_rotation_scale(
            dalaran::RotationAxisAngle(
                {0.0f, 0.0f, 1.0f},
                dalaran::Angle::radians(TAU / 8.0f)
            ),
            2.0f
        )
    );
    rec.log("base/rotated_scaled", arrow);
}

File: docs/snippets/all/archetypes/transform3d_simple.py
"""Log different transforms between three arrows."""

from math import pi

import dalaran as dl
from dalaran.datatypes import Angle, RotationAxisAngle

dl.init("dalaran_example_transform3d", spawn=True)

arrow = dl.Arrows3D(origins=[0, 0, 0], vectors=[0, 1, 0])

dl.log("base", arrow)

dl.log("base/translated", dl.Transform3D(translation=[1, 0, 0]))
dl.log("base/translated", arrow)

dl.log(
    "base/rotated_scaled",
    dl.Transform3D(
        rotation=RotationAxisAngle(axis=[0, 0, 1], angle=Angle(rad=pi / 4)),
        scale=2,
    ),
)
dl.log("base/rotated_scaled", arrow)
